integration/specs/InternalOverlay/internalOverlay-1.spec.js

```diff
@@ -31,4 +46,53 @@ describe('InternalOverlay with default position resolver', () => {
         $('[data-preview="InternalOverlay"]').click();
         expect(internalOverlay.exists()).toBe(false);
     });
+
+    it('should set focus on trigger when press tab key', () => {
+        const button = new PageInternalOverlayButton(BUTTON);
+        $('[data-preview="InternalOverlay"]').click();
+        browser.keys(TAB_KEY);
+        expect(button.hasFocusTrigger()).toBe(true);
+    });
+
+    it('should exist after set focus on trigger and press enter key', () => {
+        const button = new PageInternalOverlayButton(BUTTON);
+        const internalOverlay = new PageInternalOverlay(OVERLAY);
+        $('[data-preview="InternalOverlay"]').click();
+        browser.keys(TAB_KEY);
+        expect(button.hasFocusTrigger()).toBe(true);
+        browser.keys(ENTER_KEY);
+        expect(internalOverlay.exists()).toBe(true);
+    });
+
+    it('should not exist after set focus on trigger and press enter key twice', () => {
+        const button = new PageInternalOverlayButton(BUTTON);
+        const internalOverlay = new PageInternalOverlay(OVERLAY);
+        $('[data-preview="InternalOverlay"]').click();
+        browser.keys(TAB_KEY);
+        expect(button.hasFocusTrigger()).toBe(true);
+        browser.keys(ENTER_KEY);
+        browser.keys(ENTER_KEY);
+        expect(internalOverlay.exists()).toBe(false);
+    });
+
+    it('should exist after set focus on trigger and press space key', () => {
+        const button = new PageInternalOverlayButton(BUTTON);
+        const internalOverlay = new PageInternalOverlay(OVERLAY);
+        $('[data-preview="InternalOverlay"]').click();
+        browser.keys(TAB_KEY);
+        expect(button.hasFocusTrigger()).toBe(true);
+        browser.keys(SPACE_KEY);
+        expect(internalOverlay.exists()).toBe(true);
+    });
+
+    it('should not exist after set focus on trigger and press space key twice', () => {
+        const button = new PageInternalOverlayButton(BUTTON);
+        const internalOverlay = new PageInternalOverlay(OVERLAY);
+        $('[data-preview="InternalOverlay"]').click();
+        browser.keys(TAB_KEY);
+        expect(button.hasFocusTrigger()).toBe(true);
+        browser.keys(SPACE_KEY);
+        browser.keys(SPACE_KEY);
+        expect(internalOverlay.exists()).toBe(false);
+    });
```

Comment:
I think we should keep only the two first test, simply checking once that is open and close, removing the button page object, and using the trigger directly in the test.
The other thing we should test is that the container render in the right position respect to the trigger element, but we need to discuss about it, think the approach to test this and any method to add to the page object and let's discuss before doing any test

Reply:
I get it

Reply:
> I think we should keep only the two first test, simply checking once that is open and close, removing the button page object, and using the trigger directly in the test.
> The other thing we should test is that the container render in the right position respect to the trigger element, but we need to discuss about it, think the approach to test this and any method to add to the page object and let's discuss before doing any test

@rgah2107 We need to add test to positioning, but first you need to make a page object methods proposal

Reply:
received

Reply:
```
getPositionRespectToTheTriggerElement(triggerElement) {
        const triggerElementBoundsRect = {
            ...triggerElement.getLocation(),
            ...triggerElement.getSize(),
        };
        const internalOverlaytBoundsRect = {
            ...$(this.rootElement).getLocation(),
            ...$(this.rootElement).getSize(),
        };
        console.log('triggerElementBoundsRect');
        console.log(triggerElementBoundsRect);
        console.log('internalOverlaytBoundsRect');
        console.log(internalOverlaytBoundsRect);
        if (
            internalOverlaytBoundsRect.x === triggerElementBoundsRect.x &&
            internalOverlaytBoundsRect.y > triggerElementBoundsRect.y
        ) {
            return 'BottomLeft';
        }
        if (
            internalOverlaytBoundsRect.x + internalOverlaytBoundsRect.width ===
                triggerElementBoundsRect.x + triggerElementBoundsRect.width &&
            internalOverlaytBoundsRect.y > triggerElementBoundsRect.y
        ) {
            return 'BottomRight';
        }
        if (
            internalOverlaytBoundsRect.x < triggerElementBoundsRect.x &&
            internalOverlaytBoundsRect.x + internalOverlaytBoundsRect.width >
                triggerElementBoundsRect.x + triggerElementBoundsRect.width &&
            internalOverlaytBoundsRect.y > triggerElementBoundsRect.y
        ) {
            return 'BottomCenter';
        }
        if (
            internalOverlaytBoundsRect.x === triggerElementBoundsRect.x &&
            internalOverlaytBoundsRect.y < triggerElementBoundsRect.y
        ) {
            return 'UpLeft';
        }
        if (
            internalOverlaytBoundsRect.x + internalOverlaytBoundsRect.width ===
                triggerElementBoundsRect.x + triggerElementBoundsRect.width &&
            internalOverlaytBoundsRect.y < triggerElementBoundsRect.y
        ) {
            return 'UpRight';
        }
        if (
            internalOverlaytBoundsRect.x < triggerElementBoundsRect.x &&
            internalOverlaytBoundsRect.x + internalOverlaytBoundsRect.width >
                triggerElementBoundsRect.x + triggerElementBoundsRect.width &&
            internalOverlaytBoundsRect.y < triggerElementBoundsRect.y
        ) {
            return 'UpCenter';
        }
        if (
            internalOverlaytBoundsRect.x > triggerElementBoundsRect.x &&
            internalOverlaytBoundsRect.y < triggerElementBoundsRect.y
        ) {
            return 'CenterLeft';
        }
        if (
            internalOverlaytBoundsRect.x + internalOverlaytBoundsRect.width <
                triggerElementBoundsRect.x &&
            internalOverlaytBoundsRect.y < triggerElementBoundsRect.y
        ) {
            return 'CenterRight';
        }
        return null;
    }
```

Reply:
this could be a spec if we use the above method:
```
it('should be rendered in the right position respect to the trigger element', () => {
        const internalOverlay = new PageInternalOverlay(OVERLAY);
        const triggerButton = $(BUTTON);
        triggerButton.click();
        expect(internalOverlay.exists()).toBe(true);
        expect(internalOverlay.getPositionRespectToTheTriggerElement(triggerButton)).toBe('UpLeft');
    });
```

Reply:
in the method I assume that the dimensions of the container are always greater than the dimensions of the trigger element, as is the case in the examples

Reply:
I also believe that this method could be partitioned into one for each possible position and would return a boolean instead of a string

example:
```
isPositionedToBottomLeft(triggerElement){...}
```